src/site/content/en/mini-apps/mini-app-alternative-runtime-environments/index.md

```diff
@@ -0,0 +1,264 @@
+  </figcaption>
+</figure>
+
+## Mini apps on the desktop
+
+### Mini apps in WeChat Desktop
+
+Using the WeChat desktop client available for
+[macOS](https://dldir1.qq.com/weixin/mac/WeChatMac.dmg)
+and [Windows](https://dldir1.qq.com/weixin/Windows/WeChatSetup.exe)), it is possible to run WeChat
+mini apps on the desktop. (Make sure to _not_ load the macOS
+[version from the App Store](https://itunes.apple.com/cn/app/id836500024), since it is more limited.)
+
+To test it on macOS, share a mini app from a mobile device with yourself via the "File Transfer"
+account. This will result in a message that you can then open on the desktop client. In most cases,
+the mini app will then be directly clickable and runnable. In other cases, you have to forward the
+chat history to yourself again from a mobile device.
+
+<figure class="w-figure">
+  <img src="wechat-desktop-macos.png" alt="The WeChat macOS desktop client showing a chat with oneself with a shared mini app and a chat history as the two visible messages.">
+  <figcaption class="w-figure">
+    Sharing a mini app with oneself in the WeChat macOS desktop client.
+  </figcaption>
+</figure>
+
+On Windows, the workaround to share mini apps with oneself is not necessary, since there is a
+dedicated mini apps panel that shows the user's recently used mini apps and also includes an app
+search where new mini apps can be discovered.
+
+<figure class="w-figure">
+  <img src="wechat-desktop-windows.png" alt="The mini app panel in the WeChat Windows client showing the user's recently used mini apps.">
+  <figcaption class="w-figure">
+    The mini app panel in the WeChat Windows client.
+  </figcaption>
+</figure>
+
+<figure class="w-figure">
+  <img src="wechat-desktop-search.png" alt="The mini app search in the WeChat Windows client showing mini apps listed in various categories like games, business, education, etc.">
+  <figcaption class="w-figure">
+    The mini app search in the WeChat Windows client.
+  </figcaption>
+</figure>
+
+WeChat mini apps on the desktop naturally integrate with the operating system. On both macOS and
+Windows, they get their own entry in the multitasking bar and have their own taskbar icon. While on
+macOS, there is an option to be kept in the Dock, the icon disappears the moment the WeChat client
+app gets closed. On Windows, mini app icons can be pinned to the taskbar, but cannot be launched. On
+macOS, the title of the app is always "WeChat" and not the actual title of the app, whereas the
+title is displayed correctly on Windows.
+
+<figure class="w-figure">
+  <img src="macos-multitask.png" alt="The macOS multitask switcher includes mini apps alongside regular macOS app.">
+  <figcaption class="w-figure">
+    The Starbucks app is a mini app and can be multitasked to like any regular macOS app.
+  </figcaption>
+</figure>
+
+<figure class="w-figure">
+  <img src="macos-dock.png" alt="The Starbucks mini app icon on the macOS Dock with a WeChat title.">
```

Comment:
I'll let this slide but we're not supposed to use other brands. It could come back to bite us.

Reply:
One thing I'm realizing now is that we're talking a lot about Chinese brands. Alipay, Tencent, ByteDance, etc. Obviously this is very necessary in order to understand mini apps. But the same reasoning that I mentioned above applies. I guess we'll just have to take the risk 🤷

Reply:
Whatever we talk about these brands is neutral and facts-based, so I guess we're fine.